integration-tests/opentelemetry-vertx/src/test/java/io/quarkus/it/opentelemetry/vertx/SqlClientTest.java

```diff
@@ -86,25 +56,35 @@ void sqlClient() {
                 .then()
                 .statusCode(HTTP_OK);
 
-        await().atMost(5, TimeUnit.SECONDS).until(() -> inMemorySpanExporter.getFinishedSpanItems().size() == 2);
+        // Why 3 spans:
+        // Table creation + HTTP + Query = 3
+
+        await().atMost(5, TimeUnit.SECONDS).until(() -> inMemorySpanExporter.getFinishedSpanItems().size() == 3);
         List<SpanData> spans = inMemorySpanExporter.getFinishedSpanItems();
-        assertEquals(2, spans.size());
-
-        assertEquals(spans.get(0).getTraceId(), spans.get(1).getTraceId());
-        assertEquals(spans.get(0).getSpanId(), spans.get(1).getParentSpanId());
-
-        assertEquals("GET /sqlClient", spans.get(0).getName());
-        assertEquals(HTTP_OK, spans.get(0).getAttributes().get(HTTP_STATUS_CODE));
-
-        assertEquals("SELECT USERS", spans.get(1).getName());
-        assertEquals(CLIENT, spans.get(1).getKind());
-        assertEquals("h2", spans.get(1).getAttributes().get(DB_SYSTEM));
-        assertEquals("SELECT", spans.get(1).getAttributes().get(DB_OPERATION));
-        assertEquals("SELECT * FROM USERS", spans.get(1).getAttributes().get(DB_STATEMENT));
-        assertEquals("quarkus", spans.get(1).getAttributes().get(DB_USER));
-        assertNotNull(spans.get(1).getAttributes().get(DB_CONNECTION_STRING));
-        //noinspection ConstantConditions
-        assertTrue(spans.get(1).getAttributes().get(DB_CONNECTION_STRING).startsWith("jdbc:h2:tcp://localhost"));
+        assertEquals(3, spans.size());
+
+        inMemorySpanExporter.getFinishedSpanItems().forEach(sd -> {
+            System.out.println(sd);
+        });
+
+        // We cannot rely on the order, we must identify the spans.
+        SpanData httpSpan = inMemorySpanExporter.getFinishedSpanItems().stream()
+                .filter(sd -> sd.getName().contains("GET /sqlClient")).findFirst().orElseThrow();
+        SpanData querySpan = inMemorySpanExporter.getFinishedSpanItems().stream()
+                .filter(sd -> sd.getName().contains("SELECT USERS")).findFirst().orElseThrow();
```

Comment:
There is a 3rd span and it's not asserted. What is it?

Reply:
It's the span for the table creation. Not sure we need to assert it. There are no relations with the other spans (maybe that's what we need to check).

Reply:
makes sense

Reply:
Done.